nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AttributesToCSV.java

```diff
@@ -0,0 +1,272 @@
+import org.apache.nifi.annotation.behavior.WritesAttribute;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.flowfile.attributes.CoreAttributes;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessorInitializationContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+@EventDriven
+@SideEffectFree
+@SupportsBatching
+@Tags({"csv", "attributes", "flowfile"})
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@CapabilityDescription("Generates a CSV representation of the input FlowFile Attributes. The resulting CSV " +
+        "can be written to either a newly generated attribute named 'CSVAttributes' or written to the FlowFile as content.  " +
+        "If the attribute value contains a comma, newline or double quote, then the attribute value will be " +
+        "escaped with double quotes.  Any double quote characters in the attribute value are escaped with " +
+        "another double quote.  If the attribute value does not contain a comma, newline or double quote, then the " +
+        "attribute value is returned unchanged.")
+@WritesAttribute(attribute = "CSVAttributes", description = "CSV representation of Attributes")
+public class AttributesToCSV extends AbstractProcessor {
+
+    private static final String OUTPUT_NEW_ATTRIBUTE = "flowfile-attribute";
+    private static final String OUTPUT_OVERWRITE_CONTENT = "flowfile-content";
+    private static final String OUTPUT_ATTRIBUTE_NAME = "CSVAttributes";
+    private static final String OUTPUT_SEPARATOR = ",";
+    private static final String OUTPUT_MIME_TYPE = "text/csv";
+
+
+    public static final PropertyDescriptor ATTRIBUTES_LIST = new PropertyDescriptor.Builder()
+            .name("attribute-list")
+            .displayName("Attribute List")
+            .description("Comma separated list of attributes to be included in the resulting CSV. If this value " +
+                    "is left empty then all existing Attributes will be included. This list of attributes is " +
+                    "case sensitive and does not support attribute names that contain commas. If an attribute specified in the list is not found it will be emitted " +
```

Comment:
why not use http://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/StringEscapeUtils.html as you do with values for the attributes and headers?

Reply:
will do, thanks for the feedback.